test_conformance/api/test_mem_object_properties_queries.cpp

```diff
@@ -0,0 +1,302 @@
+    const size_t region[3] = { size_x, size_y, 1 };
+
+    std::vector<cl_uint> src_data(size);
+    std::vector<cl_uint> dst_data(size);
+    d = init_genrand(gRandomSeed);
+
+    generate_random_data(kUInt, size, d, src_data.data());
+    generate_random_data(kUInt, size, d, dst_data.data());
+    free_mtdata(d);
+    d = NULL;
+    const char* kernel_src = test_case.src.c_str();
+    error =
+        create_single_kernel_helper(context, &program, &kernel, 1, &kernel_src,
+                                    test_case.kernel_name.c_str());
+
+    test_error(error, "create_single_kernel_helper failed");
+
+    flags = (cl_mem_flags)(CL_MEM_READ_ONLY | CL_MEM_COPY_HOST_PTR);
+    error = create_object_and_check_properties(context, obj_src, test_case,
+                                               flags, src_data, size_x, size_y);
+    test_error(error, "create_object_and_check_properties obj_src failed.");
+
+    flags = (cl_mem_flags)(CL_MEM_WRITE_ONLY | CL_MEM_COPY_HOST_PTR);
+    error = create_object_and_check_properties(context, obj_dst, test_case,
+                                               flags, dst_data, size_x, size_y);
+    test_error(error, "create_object_and_check_properties obj_dst failed.");
+
+    error = clSetKernelArg(kernel, 0, sizeof(obj_src), &obj_src);
+    test_error(error, "clSetKernelArg 0 failed.");
+
+    error = clSetKernelArg(kernel, 1, sizeof(obj_dst), &obj_dst);
+    test_error(error, "clSetKernelArg 1 failed.");
+    if (test_case.obj_t == image)
+    {
+        error = clEnqueueNDRangeKernel(queue, kernel, 2, NULL, global_dim, NULL,
+                                       0, NULL, &event);
+        test_error(error, "clEnqueueNDRangeKernel failed.");
+
+        error = clWaitForEvents(1, &event);
+        test_error(error, "clWaitForEvents failed.");
+
+        error = clEnqueueReadImage(queue, obj_dst, CL_TRUE, origin, region, 0,
+                                   0, dst_data.data(), 0, NULL, NULL);
+        test_error(error, "clEnqueueReadImage failed.");
+    }
+    if (test_case.obj_t == buffer)
+    {
+        error = clEnqueueNDRangeKernel(queue, kernel, 1, NULL, &size, NULL, 0,
+                                       NULL, &event);
+        test_error(error, "clEnqueueNDRangeKernel failed.");
+
+        error = clWaitForEvents(1, &event);
+        test_error(error, "clWaitForEvents failed.");
+
+        error = clEnqueueReadBuffer(queue, obj_dst, CL_TRUE, 0,
+                                    dst_data.size() * sizeof(cl_uint),
+                                    dst_data.data(), 0, NULL, NULL);
+        test_error(error, "clEnqueueReadBuffer failed.");
+    }
+
+    for (size_t i = 0; i < size; ++i)
+    {
+        if (dst_data[i] != src_data[i])
+        {
+            log_error("ERROR: Output results mismatch.");
+            return TEST_FAIL;
+        }
+    }
+
+    return 0;
+}
+
+
+int test_image_properties_queries(cl_device_id deviceID, cl_context context,
```

Comment:
The test should check that `CL_DEVICE_IMAGE_SUPPORT` is `CL_TRUE` before proceeding with the rest of the test case.

Reply:
fixed